src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala

```diff
@@ -1,48 +1,110 @@
       extractRequest { request =>
-        cookie(tokenCookieName) { tokenCookie => // rejected with MissingCookieRejection if the cookie is not present
+        extractUserInfo { userInfo => // rejected with AuthorizationFailedRejection if the token is not present
           complete {
-            proxyService.getCachedUserInfoFromToken(tokenCookie.value).flatMap { userInfo =>
-              // Proxy logic handled by the ProxyService class
-              proxyService.proxyLocalize(userInfo, GoogleProject(googleProject), ClusterName(clusterName), request, tokenCookie)
-            }
+            // Proxy logic handled by the ProxyService class
+            proxyService.proxyLocalize(userInfo, GoogleProject(googleProject), ClusterName(clusterName), request)
           }
         }
       }
     } ~
     pathPrefix("notebooks" / Segment / Segment) { (googleProject, clusterName) =>
       extractRequest { request =>
-        cookie(tokenCookieName) { tokenCookie => // rejected with MissingCookieRejection if the cookie is not present
+        extractUserInfo { userInfo => // rejected with AuthorizationFailedRejection if the token is not present
           complete {
-            proxyService.getCachedUserInfoFromToken(tokenCookie.value).flatMap { userInfo =>
-              // Proxy logic handled by the ProxyService class
-              proxyService.proxyNotebook(userInfo, GoogleProject(googleProject), ClusterName(clusterName), request, tokenCookie)
-            }
+            // Proxy logic handled by the ProxyService class
+            proxyService.proxyNotebook(userInfo, GoogleProject(googleProject), ClusterName(clusterName), request)
           }
         }
       }
     }
 
+  /**
+    * Extracts the user token from either a cookie or Authorization header.
+    */
+  private def extractToken: Directive1[String] = {
+    optionalCookie(tokenCookieName) flatMap {
+      // We have a cookie, we're done
+      case Some(cookie) => provide(cookie.value)
+
+      // We don't have a cookie; check the Authorization header
+      case None => optionalHeaderValueByType[`Authorization`](()) flatMap {
+
+        // We have an Authorization header, extract the token
+        case Some(header) => provide(header.credentials.token)
+
+        // Not found in cookie or Authorization header, fail
+        case None => reject(AuthorizationFailedRejection)
+      }
+    }
+  }
+
+  /**
+    * Extracts the user token from the request, and looks up the cached UserInfo.
+    */
+  private def extractUserInfo: Directive1[UserInfo] = {
+    extractToken.flatMap { token =>
+      onSuccess(proxyService.getCachedUserInfoFromToken(token))
+    }
+  }
+
+  /**
+    * Sets a token cookie in the HTTP response.
+    */
+  private def setTokenCookie(userInfo: UserInfo): Directive0 = {
+    setCookie(buildCookie(userInfo))
+  }
+
+  private def buildCookie(userInfo: UserInfo): HttpCookie = {
+    HttpCookie(
```

Comment:
I made educated guesses on the cookie configuration -- hoping to validate this using SWAT